arm9/source/utils/gameutil.c

```diff
@@ -1227,24 +1227,23 @@ u32 BuildCiaFromTmdFileBuffered(const char* path_tmd, const char* path_cia, bool
     if (!name_content) return 1; // will not happen
     name_content++;
     
+    bool present[TMD_MAX_CONTENTS];
+    memset(present, true, sizeof(present));
+
     // DLC? Check for missing contents first!
     if (dlc) for (u32 i = 0; (i < content_count) && (i < TMD_MAX_CONTENTS);) {
         FILINFO fno;
         TmdContentChunk* chunk = &(content_list[i]);
         snprintf(name_content, 256 - (name_content - path_content),
             (cdn) ? "%08lx" : (dlc && !cdn) ? "00000000/%08lx.app" : "%08lx.app", getbe32(chunk->id));
-        if ((fvx_stat(path_content, &fno) != FR_OK) || (fno.fsize != (u32) getbe64(chunk->size)))
-            memmove(chunk, chunk + 1, ((--content_count) - i) * sizeof(TmdContentChunk));
-        else i++;
-    }
-    if (!content_count) return 1;
```

Comment:
This check may still make sense and shouldn't be removed.

Reply:
The check also happens near the top of the function; the difference is now, content_count isn't modified, so it doesn't need to be repeated.